website/docs/basics.md

```diff
@@ -0,0 +1,267 @@
+
+Every is managed by a [provider](#providers) which is a plugin that
+provides the implementation details. If not specified explicitly, the default provider is used.
+Providers can be configured using [provider configuration](https://www.pulumi.com/docs/concepts/resources/providers/#explicit-provider-configuration).
+
+#### Inputs and Outputs
+
+Inputs and Outputs are the primary [asynchronous data types in Pulumi](https://www.pulumi.com/docs/concepts/inputs-outputs/), 
+and they signify values that will be provided by the engine later, when the resource is created and its properties can be fetched.
+`Input[A]` type is an alias for `Output[A]` type used by [resource](#resources) arguments.
+
+Outputs are values of type `Output[A]` and behave very much like [monads](https://en.wikipedia.org/wiki/Monad_(functional_programming)). 
+This is necessary because output values are not fully known until the infrastructure resource has actually completed 
+provisioning, which happens asynchronously after the program has finished executing.
+
+Outputs are used to:
+- automatically captures dependencies between [resources](#resources)
+- provide a way to express transformations on its value before it's known
+- deffer the evaluation of its value until it's known
+- track the _secretness_ of its value
+
+Output transformations available in Besom:
+- [`map` and `flatMap`](apply_methods) methods take a callback that receives the plain value, and computes a new output
+- [lifting](lifting) directly read properties off an output value
+- [interpolation](interpolator) concatenate string outputs with other strings directly
+
+To create an output from a plain value, use the `Output` constructor, e.g.:
+```scala
+val hello = Output("hello")
+val world = Output.secret("world")
+```
+
+If you have multiple outputs of the same type and need to use them together **as a list** you can use 
+`Output.sequence` method to combine them into a single output:
+
+```scala
+val port: Output[String] = pod.name
+val host: Output[String] = node.hostname
+val hello = List(host, port).sequence // we use the extension method here
+```
+
+If you have multiple outputs of different types and need to use them together **as a tuple** you can use the standard 
+[`zip`](https://scala-lang.org/api/3.x/scala/collection/View.html#zip-1dd) method and pattern matching (`case`) to combine them into a single output:
+
+```scala
+val port: Output[Int] = pod.port
+val host: Output[String] = node.hostname
+val hello = host.zip(port).map { case (a, b) => s"https://$hostname:$port/" }
+```
+
+If you have a map of outputs and need to use them together **as a map** you can use
+`Output.traverse` method to combine them into a single output:
+
+```scala
+val m: Map[String, Output[String]] = Map(pod.name -> pod.port)
+val o: Output[Map[String, String]] = m.traverse // we use the extension method here
```

Comment:
minor: Hmmm, I think that it's a very unconventional way of using `treverse`. A `List` example will be more illustrative.

Reply:
I'm sorry, I'm not sure hot to improve this line. I'll leave as-is in the interest of delivering this content to the users ASAP. Please let me know how can I make this line better.

Reply:
I would suggest sth like:
```scala
val names: List[String] = List("John", "Paul")
val outputNames: Output[List[String]] = names.traverse(name => Output(name))
```

Reply:
Actually, the map example is what users might be looking for (other SDKs tend to support them), but I didn't I can use traverse like that so this is also very useful, thank you